packages/node_modules/samples/browser-call-with-screenshare/app.js

```diff
@@ -10,50 +10,45 @@
 /* eslint-disable arrow-body-style */
 /* eslint-disable max-len */
 
-// Declare some globals that we'll need throughout
-let activeMeeting, remoteShareStream, webex;
 
-// First, let's wire our form fields up to localStorage so we don't have to
-// retype things everytime we reload the page.
+// This is one way you might programattically determine your redirect_uri
+// depending on where you've deployed your app, but you're probably better off
+// having development/staging/production builds and injecting directly from the
+// environment.
+let redirect_uri = `${window.location.protocol}//${window.location.host}`;
 
-[
-  'access-token',
-  'invitee'
-].forEach((id) => {
-  const el = document.getElementById(id);
+if (window.location.pathname) {
+  redirect_uri += window.location.pathname;
+}
 
-  el.value = localStorage.getItem(id);
-  el.addEventListener('change', (event) => {
-    localStorage.setItem(id, event.target.value);
-  });
+let activeMeeting, remoteShareStream;
+let isPaste;
+let isOAuth;
+
+// eslint-disable-next-line no-multi-assign
+let webex = window.webex = Webex.init({
```

Comment:
Why do you initialize the SDK here before you know which way you're authenticating?

Reply:
Micheal and I decided to initialize the SDK beforehand because we defaulting to OAuth in the sample app. In addition, we had a few initializing problems when switching between OAuth and paste token method, which lead to this solution. Let me know if you need any more clarity on this and I will respond asap!

Reply:
It needs to be initialized automatically when doing the OAuth flow, especially around when it returns from OAuth. The SDK watches for the token to come across in the browser url hash. 

We could save a cookie on the auth type and only initialize if the cookie exists that the user was doing oauth mode specifically. @lalli-flores thoughts?

Reply:
@adamweeks, I think I like that idea better of storing the auth type in a cookie or local storage

Reply:
Detect if this is being called from a redirect and initialize the SDK.
``` javascript
const params = new URLSearchParams(window.location.search);
if (params.has('code') && params.has('state')) {
  // Initialize SDK
}
```
Most likely if `state` and `code` are in the query string, this was a redirect from CI.